src/Http/Http/src/BindingAddress.cs

```diff
@@ -88,7 +97,14 @@ public static BindingAddress Parse(string address)
             }
             else
             {
-                pathDelimiterStart = address.IndexOf(":", schemeDelimiterEnd + UnixPipeHostPrefix.Length, StringComparison.Ordinal);
+                var unixPipeHostPrefixLength = UnixPipeHostPrefix.Length;
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    // Windows has drive letters and volume separator (c:)
+                    unixPipeHostPrefixLength += 2;
```

Comment:
Should we verify that the characters between `schemeDelimiterEnd`, and `schemeDelimiterEnd + unixPipeHostPrefixLength` actually matches `[A-Za-z]:` like we expect? What happens now someone tried to use a relative path to a single letter file in the PWD?

Reply:
Good point, we should definitely verify it. I realized my assumptions were missing. Theoretically, we may also open an endpoint on a network shared folder like "\\COMPUTER-NAME\test". I'll research more.

Reply:
Now, I added a length check and also it's checked that socket path is rooted or not on all operating systems.